src/server/pps/server/api_server.go

```diff
@@ -341,7 +341,22 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 		}
 		outputCommit, err = pfsAPIClient.StartCommit(ctx, startCommitRequest)
 		if err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), "already exists") {
+				if _, err := pfsAPIClient.DeleteCommit(
+					ctx,
+					&pfsclient.DeleteCommitRequest{
+						Commit: client.NewCommit(parent.Repo.Name, strings.Split(parent.ID, "/")[0]),
```

Comment:
I'm a little worried about calling `DeleteCommit` with a branch name.  Could hit some weird race conditions where we end up deleting a commit different from the one we intended to delete.  Would it be hard to use the parent ID to compute the real commit ID (e.g. `master/3`) of the commit we are trying to create?